web/src/components/Plugin/PluginPage.tsx

```diff
@@ -280,11 +284,10 @@ const PluginPage: React.FC<Props> = ({
         const newPlugins = {
           ...initialData,
           [name]: { ...monacoData, disable: !formData.disable },
+          ...(shouldDelete ? { [name]: null } : {}),
         };
-        if (shouldDelete === true) {
-          newPlugins[name] = null;
-        }
-        onChange(newPlugins, form.getFieldValue('plugin_config_id'));
+        const handleType = shouldDelete ? 'delete' : 'edit';
+        onChange(newPlugins, handleType, form.getFieldValue('plugin_config_id'));
```

Comment:
Please check the code here, I think it is the data there that is having an effect, we may not need to pass the `initialData` data. The onChange logic here I don't think should be changed, but the source of the problem should be looked at.
https://github.com/apache/apisix-dashboard/blob/master/web/src/pages/Plugin/PluginMarket.tsx#L54

Reply:
Well, based on your hints, I think I've found the root cause，thank you.